Calibration complete!

jmoldon opened the pull request:
Relevant points:

1. Now we have all the steps ready:  all phase, amplitude, bandpass, bandpass with spectral index, and flux scale calibration can be conducted.

2. New global system to manage calibration tables. A dictionary is kept with all the calibration tables produced and which sources should be used to apply them. There are no ambiguities and calibration can be tracked, recomputed and applied at any stage easily.

3. There is the option now to average the data after autoflagging.

4. There is an applycal step at the end. Also, each step of the calibration can be applied by selecting "2" in the inputs file (0=don't run step, 1=run step, 2=run step and apply calibration up to that point).

5. dfluxpy still not implemented to set the flux scale. Now just a reference value is used.

6. The pipeline now produces data that is ready to image, and works well. However, AOFlagger alone is not enough to produce perfect data. I have scripts to apply additional automatic flagging. In particular using TFCROP within CASA, and works pretty well. The idea is to apply an initial bandpass and flag on bandpass-corrected data. I like the results, so maybe in the end I will avoid using aoflagger at all. All that needs to be implemented yet.


[This branch has some mismatch with respect to the current master, I think I didn't have the smooth_bp changes in my local master. But it doesn't matter, this version is the new one and it in fact contains/improves anything done in smooth_bp.]

jradcliffe5 commented:
Fantastic, this is awesome Javier.

Will run it on some eMERLIN data soon to try it out. Apologies for not working on this as much due to other commitments.

Eventually I will update the GUI.

Cheers,
Jack

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Jack Radcliffe

Room 153, Kapteyn Astronomical Institute,
University of Groningen
The Netherlands

Web:    http://www.astro.rug.nl/~radcliff
E-mail: j.f.radcliffe@astro.rug.nl<mailto:j.f.radcliffe@astro.rug.nl> / jack.radcliffe@manchester.ac.uk<mailto:jack.radcliffe@manchester.ac.uk>
Tel:      +44 (0)7950524280

Also affiliated with the University of Manchester
and the Netherlands Institute for Radio Astronomy (ASTRON)

On 22 Aug 2017, at 18:53, Javier Moldon <notifications@github.com<mailto:notifications@github.com>> wrote:


Relevant points:

  1.  Now we have all the steps ready: all phase, amplitude, bandpass, bandpass with spectral index, and flux scale calibration can be conducted.

  2.  New global system to manage calibration tables. A dictionary is kept with all the calibration tables produced and which sources should be used to apply them. There are no ambiguities and calibration can be tracked, recomputed and applied at any stage easily.

  3.  There is the option now to average the data after autoflagging.

  4.  There is an applycal step at the end. Also, each step of the calibration can be applied by selecting "2" in the inputs file (0=don't run step, 1=run step, 2=run step and apply calibration up to that point).

  5.  dfluxpy still not implemented to set the flux scale. Now just a reference value is used.

  6.  The pipeline now produces data that is ready to image, and works well. However, AOFlagger alone is not enough to produce perfect data. I have scripts to apply additional automatic flagging. In particular using TFCROP within CASA, and works pretty well. The idea is to apply an initial bandpass and flag on bandpass-corrected data. I like the results, so maybe in the end I will avoid using aoflagger at all. All that needs to be implemented yet.

[This branch has some mismatch with respect to the current master, I think I didn't have the smooth_bp changes in my local master. But it doesn't matter, this version is the new one and it in fact contains/improves anything done in smooth_bp.]

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47

Commit Summary

  *   Bandpass completed. Amplitude solutions now smoothed with smooth cal. Simpler functions call: run_gaincal, run_badpass.
  *   Changed 'vis' to 'msfile' to avoid any conflict with names.
  *   New caltables as a dictionary, saved in a file. New function run_gaincal works with caltables. solve_delays implemented
  *   Now solve_delays uses the caltables implementation
  *   Now initial_bp_cal uses the caltables implementation.
  *   Some cleaning.
  *   The clean models are included so it is not needed to untar the files
  *   Included run_initialize_models to set 3C286 model and initialize to (1,0) the other sources.
  *   Included initial_gaincal to solve for phase and amp&phase for all calibrators, and scan-averaged phase for the phase calibrator.
  *   Included eM_fluxscale to do the flux density scale calibration. Still NOT implemented dfluxpy to find the scaling factor.
  *   Included bandpass_sp to recompute the BP table including spectral index information.
  *   Calibration complete. Now included run_applycal although it needs more work
  *   Added setToCasaOrder to importfitsidi by default
  *   Added sources dictionary and applycal step.
  *   Amend some bugs
  *   Included run_split to average data after autoflagging.
  *   Added option (2 in inputs file) to apply calibration at different stages

File Changes

  *   A calibrator_models/1331+305.clean.model.tt0/logtable/table.dat<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-0> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/logtable/table.f0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-1> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/logtable/table.info<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-2> (4)
  *   A calibrator_models/1331+305.clean.model.tt0/logtable/table.lock<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-3> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/table.dat<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-4> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/table.f0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-5> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/table.f0_TSM0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-6> (0)
  *   A calibrator_models/1331+305.clean.model.tt0/table.info<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-7> (3)
  *   A calibrator_models/1331+305.clean.model.tt0/table.lock<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-8> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/logtable/table.dat<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-9> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/logtable/table.f0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-10> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/logtable/table.info<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-11> (4)
  *   A calibrator_models/3C286_C.clean.model.tt0/logtable/table.lock<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-12> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/table.dat<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-13> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/table.f0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-14> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/table.f0_TSM0<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-15> (0)
  *   A calibrator_models/3C286_C.clean.model.tt0/table.info<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-16> (3)
  *   A calibrator_models/3C286_C.clean.model.tt0/table.lock<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-17> (0)
  *   M eMERLIN_CASA_pipeline.py<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-18> (173)
  *   M functions/eMERLIN_CASA_functions.py<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-19> (586)
  *   M inputs.txt<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47/files#diff-20> (8)

Patch Links:

  *   https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47.patch
  *   https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/e-merlin/CASA_eMERLIN_pipeline/pull/47>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ALNaUJNTsot-EDsgsO6SjdFRDsp3ZAK_ks5saweggaJpZM4O-5zE>.

daniellefenech commented:
Hi Javier,

Apologies for being so quiet on this for a while. That's great news and I'm going to try and run this through on some C-band data this week, will let you know how it goes.

I have a couple of questions: you mentioned about potentially dropping the use of AOFlagger in favour of the CASA flagging tools. I have spent some time sorting a CASA version of SERPent and hoping for this to be finished in the next week or two (maybe for some fiddling at the workshop?) i also think we decided we were going to aim for getting a CASA-friendly version of SERBob going as well. Given you've been more in the thick of this, do you still think that's something we're likely to want/need? 

The second is do you want me to take a look at updating dfluxpy or has this already been started?

jmoldon commented:
Hi Danielle,

Great that you are ready to test it. This is the branch that you need to use. I think it is better to merge this branch to master now, but I cannot do it (I updated the settings so a different reviewer needs to accept the merge). So please you or @jradcliffe5 complete this pull request so anyone can try the pipeline from the master branch.

I have tried it with different types of data sets: C, L band, with multiple pairs of target/phasecals, with mixed mode data, etc. and seems to work in all cases. For sure new bugs will appear, but now is a good milestone to have a common master version for all.

Regarding flagging, I made some tests to compare AOFlagger vs CASA (tfcrop) and I liked the results of TFCROP: it has an iterative procedure to refine a bandpass shape with different segments similar to what Bob was doing, but probably more accurate. Using that, I have seen that OQ208 can be flagged well, so I found a BP table, corrected all sources and flagged them again. The benefits of that with respect to AOflagger only are:
1. you can easily find flags from partially calibrated data, which is very useful, and I don't see an easy way to do it from aoflagger.
2. It is easier to keep track of what has been flagged, which parameters were used, and get some statistics.
I have not tried these yet, but:
3) In the CASA flagging modes you can specify some averaging (in time and freq) that will help to find low-amplitude RFIs at different stages of the calibration or at the end.
4) Once the calibration is done, RFLAG mode can be used to flag completely calibrated data, which I assume will be more efficient than aoflagger.

In any case, I have also seen that the aoflagger algorithm (and so probably SERBob also) is more efficient on removing very bad RFIs with completely untouched data. We have to put all the pieces together, but I can image that the best approach would be something like this:

- SERBob will be run to remove very bad RFIs from raw data efficiently but being conservative, to remove Lo drop outs, maybe also to find quack times, and any other e-MERLIN related flagging.
- We calibrate the bandpass, and flag deeper using CASA (TFCROP). If SERBob works well with partially calibrated data, we can also use it here.
- Complete the calibration and run CASA(RFLAG) with more on-the-fly averaging to remove very low remaining RFIs.

I think with that approach we can get very good automatic flagging even when keeping full-resolution data.

Let me know when you have a working version of SERBob and I can start making tests. The best would be to have the possibility to call the different flaggers from the pipeline for the meeting so we can compare different approaches there.

Also, yes, if you can implement dfluxpy to find the flux scale parameter eMfactor within the function eM_fluxscale that would be great!


Cheers,

  Javier

daniellefenech commented:
Hi Javier

Okay, it sounds like it's going to be worth continuing then. As it stands I'm finishing up a version of SERPent for CASA, which I'll get into a function so it can hopefully be included without much hassle. I think we agreed that I'd head up to Manchester the week beginning the 9th October to spend a few days in earnest getting SERBob going, so I guess it'd be after that before we'd have useable CASA version.

I'll take a look at dfluxpy as well then!

Thanks,

Danielle
